pkg/wireguard/agent/agent.go

```diff
@@ -168,60 +164,6 @@ func (a *Agent) InitLocalNodeFromWireGuard(localNode *node.LocalNode) {
 	a.optOut = localNode.OptOutNodeEncryption
 }
 
-func (a *Agent) initUserspaceDevice(linkMTU int) (netlink.Link, error) {
-	log.WithField(logfields.Hint,
-		"It is highly recommended to use the kernel implementation. "+
-			"See https://www.wireguard.com/install/ for details.").
-		Info("falling back to the WireGuard userspace implementation.")
-
-	tundev, err := tun.CreateTUN(types.IfaceName, linkMTU)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create tun device: %w", err)
-	}
-
-	uapiSocket, err := ipc.UAPIOpen(types.IfaceName)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create uapi socket: %w", err)
-	}
-
-	uapiServer, err := ipc.UAPIListen(types.IfaceName, uapiSocket)
-	if err != nil {
-		return nil, fmt.Errorf("failed to start WireGuard UAPI server: %w", err)
-	}
-
-	scopedLog := log.WithField(logfields.LogSubsys, "wireguard-userspace")
-	logger := &device.Logger{
-		Verbosef: scopedLog.Debugf,
-		Errorf:   scopedLog.Errorf,
-	}
-	dev := device.NewDevice(tundev, conn.NewDefaultBind(), logger)
-
-	// cleanup removes the tun device and uapi socket
-	a.cleanup = append(a.cleanup, func() {
```

Comment:
On closer inspection, let's also remove `a.cleanup` since it is no longer used

Reply:
Pushed https://github.com/cilium/cilium/compare/cc5725858eb8a5576a37ca386666e01f88d802d6..573368a5283dfe579798b4d91e50039acdd77deb.